crates/ursa-rpc-service/src/api.rs

```diff
@@ -97,51 +108,29 @@ where
     pub store: Arc<UrsaStore<S>>,
     pub network_send: Sender<NetworkCommand>,
     pub provider_send: Sender<ProviderCommand>,
+    pending_requests: PendingRequests,
+    client: Arc<Client>,
+    origin_config: OriginConfig,
 }
 
 #[async_trait]
 impl<S> NetworkInterface for NodeNetworkInterface<S>
 where
     S: Blockstore + Store + Send + Sync + 'static,
 {
-    async fn get(&self, cid: Cid) -> Result<Option<Vec<u8>>> {
-        if !self.store.blockstore().has(&cid)? {
-            info!("Requesting block with the cid {cid:?}");
-            let (sender, receiver) = oneshot::channel();
-            let request = NetworkCommand::GetBitswap { cid, sender };
-
-            // use network sender to send command
-            self.network_send.send(request)?;
-            if let Err(e) = receiver.await? {
-                return Err(anyhow!(
-                    "The bitswap failed, please check server logs {:?}",
-                    e
-                ));
-            }
-        }
-        self.store.blockstore().get(&cid)
+    async fn get(&self, cid: Cid) -> Result<Vec<u8>> {
+        self.sync_content(cid).await?;
+        let content =
+            self.store.blockstore().get(&cid)?.ok_or_else(|| {
```

Comment:
```suggestion
            self.store.blockstore().get(&cid)
```

Reply:
If sync content is Ok(()), error being returned early, then blockstore.get should never be empty

Reply:
"?" Should take care of that then. Not a big deal just comment

Reply:
Ah yeah unfortunately the verbosity is required since get -> Result<Option>>, so have to translate option into an error to `?` it